buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -13,39 +13,111 @@
+
+  @get:Input public abstract val currentRelease: Property<String>
+
+  @get:Input public abstract val pastRelease: Property<String>
+
+  @get:Input public abstract val printReleaseConfig: Property<String>
+
+  @OutputFile public fun getReleaseConfigFile() = project.buildDir.resolve("release.cfg")
+
+  @OutputFile public fun getReleaseReportFile() = project.buildDir.resolve("release_report.md")
+
   @TaskAction
   @Throws(Exception::class)
   fun generateReleaseConfig() {
-    val currentRelease = project.property("currentRelease").toString()
-    val pastRelease = project.property("pastRelease").toString()
-    val printReleaseConfig = project.property("printOutput").toString().toBoolean()
     val rootDir = project.rootDir
     val availableModules = parseSubProjects(rootDir)
-    val firebaseLibraries = extractLibraries(availableModules, rootDir)
 
     val repo = Git.open(rootDir)
     val headRef = repo.repository.resolve(Constants.HEAD)
-    val branchRef = getObjectRefForBranchName(repo, pastRelease)
+    val branchRef = getObjectRefForBranchName(repo, pastRelease.get())
 
-    val changedLibraries = getChangedLibraries(repo, branchRef, headRef, firebaseLibraries)
-    writeReleaseConfig(rootDir, changedLibraries, currentRelease)
-    if (printReleaseConfig) {
-      println(changedLibraries.joinToString(",", "LIBRARIES TO RELEASE: "))
+    val libsToRelease = getChangedChangelogs(project, repo, branchRef, headRef, availableModules)
+    val changedLibsWithNoChangelog =
+      Sets.difference(
+        getChangedLibraries(repo, branchRef, headRef, availableModules),
+        libsToRelease.map { it.path }.toSet()
+      )
+    val changes = getChangesForLibraries(repo, branchRef, headRef, libsToRelease)
+    writeReleaseConfig(
+      getReleaseConfigFile(),
+      ReleaseConfig(currentRelease.get(), libsToRelease.map { it.path }.toSet())
+    )
+    val releaseReport = generateReleaseReport(changes, changedLibsWithNoChangelog)
+    if (printReleaseConfig.get().toBoolean()) {
+      println(releaseReport)
     }
+    writeReleaseReport(getReleaseReportFile(), releaseReport)
   }
 
+  private fun generateReleaseReport(
+    changes: Map<String, List<CommitDiff>>,
+    changedLibrariesWithNoChangelog: Set<String>
+  ) =
+    """
+      |# Release Report
+      |${
+            changes.entries.joinToString("\n") {
+                """
+      |## ${it.key}
+      
+      |${it.value.joinToString("\n") { it.toString() }}
+      """.trimMargin()
+            }
+        }
+      |
+      |## SDKs with changes, but no changelogs
+      |${changedLibrariesWithNoChangelog.joinToString("  \n")}
+    """
+      .trimMargin()
+
+  private fun getChangesForLibraries(
+    repo: Git,
+    branchRef: ObjectId,
+    headRef: ObjectId,
+    changedLibraries: Set<Project>
+  ) =
+    changedLibraries
+      .map { getRelativeDir(it) }
+      .associateWith { getDirChanges(repo, branchRef, headRef, it) }
+      .toMap()
+
```

Comment:
I can't add a comment here, but if you update your branch- we have a new extension property `isKTXLibrary`:
```kotlin
availableModules.filterNot { it.isKTXLibrary } 
```
(assuming it's a list of projects)